Framework/DataHandling/test/LoadILLIndirect2Test.h

```diff
@@ -110,6 +110,59 @@ class LoadILLIndirect2Test : public CxxTest::TestSuite {
     TS_ASSERT(boost::ends_with(idf, "IN16BF_Definition.xml"));
   }
 
+  void test_diffraction_bats() {
+    // checks bats in the older version of IN16B diffraction mode
+    LoadILLIndirect2 loader;
+    TS_ASSERT_THROWS_NOTHING(loader.initialize())
+    TS_ASSERT(loader.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(
+        loader.setPropertyValue("Filename", m_batsDiffraction));
+    TS_ASSERT_THROWS_NOTHING(
+        loader.setPropertyValue("OutputWorkspace", "__out_ws"));
+    TS_ASSERT_THROWS_NOTHING(
+        loader.setProperty("LoadDetectors", "Diffractometer"));
+    TS_ASSERT_THROWS_NOTHING(loader.execute(););
+    TS_ASSERT(loader.isExecuted());
+    MatrixWorkspace_sptr output2D =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("__out_ws");
+    TS_ASSERT_EQUALS(output2D->getNumberHistograms(), 2049)
+    TS_ASSERT_EQUALS(output2D->blocksize(), 2048)
+
+    // check some values near the center tubes to verify the geometry
+    // used is from the older version
+    TS_ASSERT_EQUALS(output2D->dataY(1050)[1156], 16)
+    TS_ASSERT_EQUALS(output2D->dataY(871)[1157], 17)
+    TS_ASSERT_EQUALS(output2D->dataY(746)[1157], 18)
+
+    AnalysisDataService::Instance().clear();
+  }
+  void test_diffraction_doppler() {
```

Comment:
You added two tests one for doppler, one for bats. How does this relate with the old/new way of writing in nexus? Do we have data to do 2x2 =4 tests?

Reply:
No, only 2. The Doppler is in the newer version, the BATS in the older.

Reply:
OK, can you add comments in the tests to make this clear?

Reply:
I'll try to make them clearer

Reply:
Sorry, I misunderstood what you meant. Right know, the data I know of are enough for these 2 tests, but in theory we could do the 4 (but it will mainly just be an overlap of the current tests).